File: Swift/6 kyu/6kyu_Linked Lists_-_Length_&_Count.swift
class Node {
    var data: Int
    var next: Node?
    init(_ data: Int) {
        self.data = data
    }
}

func length(_ head: Node?) -> Int {
    var count = 0
    var current = head
    while current != nil {
        count += 1
        current = current?.next
    }
    return count
}

func count(_ head: Node?, _ data: Int) -> Int {
    var count = 0
    var current = head
    while current != nil {
        if current!.data == data {
            count += 1
        }
        current = current?.next
    }
    return count
}

func printLinkedList(_ head: Node?) {
    var current = head
    while current != nil {
        print(current!.data, terminator: " -> ")
        current = current?.next
    }
    print("null")
}

File: Ruby/6 kyu/6kyu_Linked Lists_-_Length_&_Count.rb
class Node
	attr_accessor :data, :next
  def initialize(data)
  	@data = data
    @next = nil
  end
end

def length(node)
  count = 0
  while node
    count += 1
    node = node.next
  end
  count
end

def count(node, data)
  count = 0
  while node
    count += 1 if node.data == data
    node = node.next
  end
  count
end